style/redux/README.md

```diff
@@ -0,0 +1,195 @@
+`createAction` is less readable and doesn't let us easily export named
+functions. Note that only `type` is mandatory.
+
+So far the common practice is to group files by type rather than by feature, and
+this style guide follows that pattern. But there's an alternative pattern -
+[ducks] - a proposal to bundle action creators with their reducers into a single
+file. Such a bundle is called a 'duck'. Its goal is to move towards a modular
+application structure.
+
+[redux-promise-middleware]: https://github.com/pburtchaell/redux-promise-middleware
+[redux-thunk]: https://github.com/gaearon/redux-thunk
+[Flux Standard Action]: https://github.com/acdlite/flux-standard-action
+[redux-actions]: https://github.com/acdlite/redux-actions
+[ducks]: https://github.com/erikras/ducks-modular-redux
+
+## Action types
+
+- Put action types in one file of constants.
+- Group related action types in an object.
+- Expose each object as a named export.
+- Expose all action types as the default export.
+
+```js
+// constants/actionTypes.js
+export const todo = {
+  fetch: 'FETCH_TODO',
+  set: 'SET_TODO',
+};
+
+export default {
+  todo,
+};
+```
+
+Action types identify the nature of the action that has occurred. Two actions
+with the same type MUST be strictly equivalent (using ===). By convention, type
+is usually a string constant or a Symbol.
+
+## Reducers
+
+- Expose reducers as the default export.
+- Expose handlers as named exports for unit testing.
+- Use constants instead of inline strings for action types.
+- Always define an `INITIAL_STATE` variable.
+
+```js
+// using redux-promise-middleware
+// using type-to-reducer
+
+// reducers/todo.js
+import typeToReducer from 'type-to-reducer';
+import { todo } from '../constants/actionTypes';
+
+const INITIAL_STATE = {
+  data: {},
+  error: false,
+  isRequesting: false,
+};
+
+export function addTodo(state, action) {
+  return { ...state, data: action.payload };
+}
+
+export const addTodoAsync = {
+  PENDING: (state) => ({
+    ...state,
+    isRequesting: true,
+  }),
+  REJECTED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    error: action.payload,
+  }),
+  FULFILLED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    data: action.payload,
+  }),
+};
+
+export default typeToReducer(
+  {
+    [todo.add]: addTodo,
+    [todo.addAsync]: addTodoAsync,
+  },
+  INITIAL_STATE
+);
+```
+
+The final reducer should be exposed as the default export. Individual handler
+functions should be exposed as named exports in order to simplify unit testing.
+
+The simplest way to create a reducer is to use the `handleActions` method of
+[redux-actions]. This avoids having to write a switch statement and a default
+handler. Another benefit is that it enforces the use of FSA-compliant action
+objects. Alternatively, use [type-to-reducer] for simplifying handling of async
+actions, as it allows nesting of related actions.
```

Comment:
I'm currently using it in oncostats, so it's not been thoroughly used.

Although I can't see how this:
```js
import { handleActions } from 'redux-actions';
import { todo } from '../constants/actionTypes';

const INITIAL_STATE = {
  data: {},
  error: false,
  isRequesting: false,
};

export function addTodo(state, action) {
  return { ...state, data: action.payload };
}

// ---------- NOTICE I'M USING handleActions INSTEAD OF typeToReducer
// So there's a lot of repetition and it's hard or not possible at all
// to reuse those 3 handlers for other similar async actions
// due to not be able to nest things here
export default handleActions({
  [todo.add]: addTodo,
  [`${todo.addAsync}_PENDING`]: (state) => ({
    ...state,
    isRequesting: true,
  }),
  [`${todo.addAsync}_REJECTED`]: (state, action) => ({
    ...state,
    isRequesting: false,
    error: action.payload,
  }),
  [`${todo.addAsync}_FULFILLED`]: (state, action) => ({
    ...state,
    isRequesting: false,
    data: action.payload,
  }),
}, INITIAL_STATE);
```

or even this: 
```js
import { todo } from '../constants/actionTypes';

const INITIAL_STATE = {
  data: {},
  error: false,
  isRequesting: false,
};

export function addTodo(state, action) {
  return { ...state, data: action.payload };
}

export default function reducer(state = INITIAL_STATE, action) => {
  // ---------- NOTICE I'M USING switch statements
  // Again, reuse is not possible here
  switch (action.type) {
    case todo.add:
      addTodo()
    case `${todo.addAsync}_PENDING`: 
      return {
        ...state,
        isRequesting: true,
      };
    case `${todo.addAsync}_REJECTED`:
      return {
        ...state,
        isRequesting: false,
        error: action.payload,
      };
    case `${todo.addAsync}_FULFILLED`:
      return {
        ...state,
        isRequesting: false,
        data: action.payload,
      };
    default:
      return state;
  }
}
```

are easier to read or manage.
There's a lot of repetition and it get's hard to read (group visually) related/derived/suffixed actions.
What do you think about this?